### PR TITLE
feat(containerdisk): Add CentOS Stream 10 containerdisk

### PR DESCRIPTION
**Pull request by jcanocan**
**What this PR does / why we need it**:

It adds support for the CentOS Stream 10 image.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # [CNV-43152](https://issues.redhat.com/browse/CNV-43152)

**Special notes for your reviewer**:

**Release note**:

```release-note
Enabled CentOS Steam 10 containerdisk
```

### CODE REVIEW

**Review by 0xFelix — COMMENTED** [review]
One nit, otherwise looks good!

**Comment by 0xFelix** [comment]
Ah, could you also add tests to `artifacts/centosstream/centos-stream_test.go`?

File: artifacts/centosstream/centos-stream.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ func (c *centos) Metadata() *api.Metadata {
 func (c *centos) Inspect() (*api.ArtifactDetails, error) {
 	var baseURL string
 
-	if strings.HasPrefix(c.Version, "8") || strings.HasPrefix(c.Version, "9") {
+	if strings.HasPrefix(c.Version, "8") || strings.HasPrefix(c.Version, "9") || strings.HasPrefix(c.Version, "10") {
```
[the remark this answers is not in the document]
Comment (jcanocan) [reply artifacts/centosstream/centos-stream.go:54]: What do you think to it in a follow-up PR? I would like to limit the scope of this PR to just add the CentOS 10.

**Comment by jcanocan** [comment]
> Ah, could you also add tests to `artifacts/centosstream/centos-stream_test.go`?

Added.

File: artifacts/centosstream/centos-stream.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (0xFelix) [reply artifacts/centosstream/centos-stream.go:54]: I'm fine with both. It would be just this one change. You can do it in a follow up as well.

**Review by 0xFelix — COMMENTED** [review]
/approve

**Review by codingben — APPROVED** [review]
/lgtm

**Pull request merged**